### PR TITLE
BUG: when flooring, ambiguous parameter unnecessarily used (and raising Error)

### LINKED ISSUE #49565: BUG: when `floor`ing, `ambiguous` parameter unnecessarily used (and raising Error)
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

ts = pd.Timestamp("2020-10-25 00:00", tz='Europe/Berlin')
# This works:
ts.floor('H') # Timestamp('2020-10-25 00:00:00+0200', tz='Europe/Berlin')
# But this gives an error:
ts.floor('H', ambiguous='infer') # ValueError: Cannot infer offset with only one time.
```

### Issue Description

The value of the `ambiguous` parameter is used by the method, even if there is no ambiguity.

### Expected Behavior

Both calls to the `.floor` method should give the same outcome.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 91111fd99898d9dcaa6bf6bedb662db4108da6e6
python           : 3.9.13.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19045
machine          : AMD64
processor        : Intel64 Family 6 Model 165 Stepping 2, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : de_DE.cp1252

pandas           : 1.5.1
numpy            : 1.23.4
pytz             : 2022.6
dateutil         : 2.8.2
setuptools       : 63.4.1
pip              : 22.2.2
Cython           : None
pytest           : 7.1.3
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 8.5.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.6.2
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by jcaoun**
- [x] closes #49565
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/scalar/timestamp/test_timezones.py [hunk 1]
```diff
@@ -355,22 +356,6 @@ def test_astimezone(self, tzstr):
         assert expected == result
         assert isinstance(result, Timestamp)
 
-    @td.skip_if_windows
-    def test_tz_convert_utc_with_system_utc(self):
-
```
Comment (MarcoGorelli) [inline pandas/tests/scalar/timestamp/test_timezones.py:360]: why delete this?

File: pandas/_libs/tslibs/timestamps.pyx [hunk 2]
```diff
@@ -1946,8 +1946,11 @@ default 'raise'
         >>> pd.NaT.tz_localize()
         NaT
         """
-        if ambiguous == "infer":
-            raise ValueError("Cannot infer offset with only one time.")
+        if not isinstance(ambiguous, bool) and ambiguous not in {"NaT", "raise"}:
+            raise ValueError(
+                        "ambiguous parameter must be one of: "
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/timestamps.pyx:1951]: let's quote the parameter name, else it'll be ambiguous (😄 )
```
                        "'ambiguous' parameter must be one of: "
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
getting there, thanks 💪 

this'll also need a whatsnew note

File: pandas/_libs/tslibs/timestamps.pyx (hunk 2 above)
Comment (jcaoun) [reply pandas/_libs/tslibs/timestamps.pyx]: done :)

File: pandas/tests/scalar/timestamp/test_timezones.py (hunk 1 above)
Comment (jcaoun) [reply pandas/tests/scalar/timestamp/test_timezones.py]: done 👍

File: doc/source/whatsnew/v1.5.3.rst [hunk 3]
```diff
@@ -25,20 +25,21 @@ Fixed regressions
 Bug fixes
 ~~~~~~~~~
 - Bug in :meth:`.Styler.to_excel` leading to error when unrecognized ``border-style`` (e.g. ``"hair"``) provided to Excel writers (:issue:`48649`)
--
+- Bug in :meth:`.Timestamp.floor` fixed tz_localize so 'ambiguous' parameter can only be True, False, 'NaT' or 'raise' and all other invalid arguments to ambiguous have same error, including 'infer' (:issue:`49565`)
```
Comment (MarcoGorelli) [inline doc/source/whatsnew/v1.5.3.rst:28]: this should go in v2.0.0.rst

as for the test, how about
```
Better error message when passing invalid values to ``ambiguous`` parameter in :meth:`Timestamp.tz_localize` (:issue:`49565`)
```

File: doc/source/whatsnew/v1.5.3.rst [hunk 4]
```diff
@@ -25,20 +25,21 @@ Fixed regressions
...
 
 .. ---------------------------------------------------------------------------
 .. _whatsnew_153.other:
 
 Other
 ~~~~~
--
+- Modified test cases in test_timezones.py to reflect new error message from bug fix for bug: (:issue:`49565`)
```
Comment (MarcoGorelli) [inline doc/source/whatsnew/v1.5.3.rst:35]: this is unnecessary, let's remove it

File: doc/source/whatsnew/v1.5.3.rst [hunk 5]
```diff
@@ -25,20 +25,21 @@ Fixed regressions
...
 -
 
 .. ---------------------------------------------------------------------------
 .. _whatsnew_153.contributors:
 
 Contributors
 ~~~~~~~~~~~~
+- Julia Chaker Aoun
```
Comment (MarcoGorelli) [inline doc/source/whatsnew/v1.5.3.rst:43]: no need, this is populated automatically

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.5.3.rst (hunk 5 above)
Comment (jcaoun) [reply doc/source/whatsnew/v1.5.3.rst:43]: Hi! If you could approve or merge my PR by tomorrow I would really appreciate it! I'm working on this for a project and I get extra credit if the PR is approved :)

Comment (MarcoGorelli) [reply doc/source/whatsnew/v1.5.3.rst:43]: will do if you address the review comments

Comment (MarcoGorelli) [reply doc/source/whatsnew/v1.5.3.rst:43]: you'll need to address the review comments first

Comment (MarcoGorelli) [reply doc/source/whatsnew/v1.5.3.rst:43]: still need to address https://github.com/pandas-dev/pandas/pull/50024?notification_referrer_id=NT_kwDOAf8KsLM0OTg0NTcwNTcxOjMzNDkxNjMy&notifications_query=reason%3Aparticipating#discussion_r1041506077 and https://github.com/pandas-dev/pandas/pull/50024?notification_referrer_id=NT_kwDOAf8KsLM0OTg0NTcwNTcxOjMzNDkxNjMy&notifications_query=reason%3Aparticipating#discussion_r1041505842

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me, I think this error message is clearer than the original (which kind of makes it look like pandas was trying to use the `ambiguous` parameter, but failed)

Over to @mroeschke in case there's further comments / objections

**Comment by jcaoun** [comment]
Thanks for approving the PR!!! This is the first time I contribute to an open-source project and it was really fun. I really appreciate your help :)

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jcaoun

**Pull request merged**